src/@context/Web3.tsx

```diff
@@ -155,16 +157,33 @@ function Web3Provider({ children }: { children: ReactNode }): ReactElement {
     if (!accountId || !networkId || !web3) return
 
     try {
+      const { oceanTokenAddress } = getOceanConfig(networkId)
+      const approvedTokensBalance: { [key: string]: string } = {}
+      const approvedTokens = (await getOpcsApprovedTokens(chainId)).filter(
```

Comment:
I see we call `getOpcsApprovedTokens` multiple times in various places, again on the pricing page. Would it be more effective if we fetch and expose approved tokens from `MarketMetadata`? Cause this Web3 context provider here can then just grab it from there, as `MarketMetadata` is our highest context provider. Cause in end we should only fetch the approved tokens once, when user initially enters app

Reply:
Yes, that is a good idea actually! I'll modify it as suggested

Reply:
In that provider I'm not going to be able to check for the selected chainId though, right? Should I expose the list for all the supported chainIds then?

Reply:
right, didn't think of that and if I count the requests current implementation could be actually most effective already. Think we're good here, sorry!

Cause if we always fetch for all chains on app start we force at least 5 or 9 requests on all users although so far they are only relevant in the publish form. But if e.g. user is connected to Polygon and publishes on Polygon we end up with 2 requests (1 in Web3 context, 1 in pricing form). We could reduce this to only 1 by storing the fetched tokens in some context but too much refactor for this PR